src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx

```diff
@@ -14,77 +18,182 @@ type Props = {
+    // when the navigation is opened, body should be scroll-locked
+    this.toggleBodyScrollLock(navVisible);
+
+    this.setState({
+      navOffsetTop: this.headerRef.current?.getBoundingClientRect().bottom ?? 0,
+      navVisible,
+    });
+  }
+
+  toggleBodyScrollLock(lock: boolean) {
+    const bodyElement = document.getElementsByTagName('body')[0];
+    window.scrollTo(0, 0);
+    bodyElement.style.overflow = lock ? 'hidden' : 'visible';
+    bodyElement.style.position = lock ? 'fixed' : 'static';
+    bodyElement.style.width = lock ? '100%' : 'auto';
+  }
+
+  render() {
+    const {params, routes, route, router, renderNavigation, children} = this.props;
+    const {navVisible, navOffsetTop} = this.state;
+
+    // We want child's view's props
+    const childProps =
+      children && React.isValidElement(children) ? children.props : this.props;
+    const childRoutes = childProps.routes || routes || [];
+    const childRoute = childProps.route || route || {};
+    const shouldRenderNavigation = typeof renderNavigation === 'function';
+
+    return (
+      <React.Fragment>
+        <SettingsColumn>
+          <SettingsHeader ref={this.headerRef}>
+            <HeaderContent>
+              {shouldRenderNavigation && (
+                <NavMenuToggle
+                  priority="link"
+                  icon={navVisible ? <IconClose /> : <IconMenu />}
+                  onClick={() => this.toggleNav(!navVisible)}
+                />
+              )}
+              <StyledSettingsBreadcrumb
+                params={params}
+                routes={childRoutes}
+                route={childRoute}
+              />
+              <SettingsSearch routes={routes} router={router} params={params} />
+            </HeaderContent>
+          </SettingsHeader>
+
+          <MaxWidthContainer>
+            {shouldRenderNavigation && (
+              <SidebarWrapper isVisible={navVisible} offsetTop={navOffsetTop}>
+                {renderNavigation!()}
+              </SidebarWrapper>
+            )}
+            <NavMask isVisible={navVisible} onClick={() => this.toggleNav(false)} />
+            <Content>{children}</Content>
+          </MaxWidthContainer>
+        </SettingsColumn>
+      </React.Fragment>
+    );
+  }
 }
 
-SettingsLayout.propTypes = {
-  renderNavigation: PropTypes.func,
-  route: PropTypes.object,
-  router: PropTypes.object,
-  routes: PropTypes.array,
-};
+const SettingsColumn = styled('div')`
+  display: flex;
+  flex-direction: column;
+  flex: 1; /* so this stretches vertically so that footer is fixed at bottom */
+  min-width: 0; /* fixes problem when child content stretches beyond layout width */
+  footer {
+    margin-top: 0;
+  }
+`;
+
+const HeaderContent = styled('div')`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const NavMenuToggle = styled(Button)`
+  display: none;
+  margin-right: ${space(2)};
+  color: ${p => p.theme.gray600};
+  &:hover,
```

Comment:
I believe you don't need `&` here

Reply:
Should we be consistent with the rest of the codebase? Everywhere else we are using `&` in front of pseudo-selectors. It is working without it, but I think we should decide on one style and use it everywhere - maybe something to talk about on our TSC?

Reply:
yes we can discuss it in the next TSC 😉